libcaf_net/caf/net/stream_socket.test.cpp

```diff
@@ -0,0 +1,113 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#include "caf/net/stream_socket.hpp"
+
+#include "caf/test/test.hpp"
+
+#include "caf/byte_buffer.hpp"
+#include "caf/log/test.hpp"
+#include "caf/span.hpp"
+
+using namespace caf;
+using namespace caf::net;
+
+namespace {
+
+template <class T>
+T unbox(caf::expected<T> x) {
+  if (!x)
+    test::runnable::current().fail("{}", to_string(x.error()));
+  return std::move(*x);
+}
+
+std::byte operator"" _b(unsigned long long x) {
+  return static_cast<std::byte>(x);
+}
+
+} // namespace
+
+namespace {
+
+struct fixture {
+  fixture() : rd_buf(124) {
+    std::tie(first, second) = unbox(make_stream_socket_pair());
+    if (auto err = nonblocking(first, true); err != caf::none)
+      CAF_RAISE_ERROR(
+        detail::format("cannot create socket with error {}", err).c_str());
+    if (auto err = nonblocking(second, true); err != caf::none)
+      CAF_RAISE_ERROR(
+        detail::format("cannot create socket with error {}", err).c_str());
+    if (auto buffer_size = unbox(send_buffer_size(first)); buffer_size == 0u)
+      CAF_RAISE_ERROR(
+        detail::format("buffer size for first: {}", buffer_size).c_str());
+    if (auto buffer_size = unbox(send_buffer_size(second)); buffer_size == 0u)
+      CAF_RAISE_ERROR(
+        detail::format("buffer size for first: {}", buffer_size).c_str());
```

Comment:
```suggestion
    require_eq(nonblocking(first, true), error{});
    require_eq(nonblocking(second, true), error{});
    require_eq(send_buffer_size(first), 0u);
    require_eq(send_buffer_size(second), 0u);
```

Reply:
We can not currently use the runnable instance before the fixture is constructed. Due to this, the `require_eq()` throws error and the test fails directly.